sklearn/model_selection/tests/test_validation.py

```diff
@@ -240,6 +241,7 @@ def fit(
                 sparse_sample_weight.shape[0], X.shape[0]
             )
         if sparse_param is not None:
+            P_sparse = type(sparse_param)(P)
```

Comment:
We don't really need `P_sparse` here. We only want to check the shape.
So you can remove this line and change `P_sparse` by `P` below.